dataflow/gemma-flex-template/custom_model_gemma.py

```diff
@@ -0,0 +1,182 @@
+          model: The Gemma model being used.
+          inference_args: Any additional arguments for an inference.
+
+        Returns:
+          An Iterable of type PredictionResult.
+        """
+        # Loop each text string, and use a tuple to store the inference results.
+        predictions = []
+        result = model.generate(prompts=batch, device=self._device)
+        predictions.append(result)
+        return [PredictionResult(x, y) for x, y in zip(batch, predictions)]
+
+
+class FormatOutput(beam.DoFn):
+    def process(self, element, *args, **kwargs):
+        yield "Input: {input}, Output: {output}".format(
+            input=element.example, output=element.inference)
+
+
+if __name__ == "__main__":
+    import argparse
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--messages_subscription",
+        required=True,
+        help="Pub/Sub subscription for input text messages",
+    )
+    parser.add_argument(
+        "--responses_topic",
+        required=True,
+        help="Pub/Sub topic for output text responses",
+    )
+    parser.add_argument(
+        "--model_variant",
+        required=False,
+        default="gemma-2b-it",
+        help="name of the gemma variant being used",
+    )
+    parser.add_argument(
+        "--checkpoint_path",
+        required=False,
+        default="pytorch_model/gemma-2b-it.ckpt",
+        help="path to the Gemma model weights in the custom worker container",
+    )
+    parser.add_argument(
+        "--tokenizer_path",
+        required=False,
+        default="pytorch_model/tokenizer.model",
+        help="path to the Gemma tokenizer in the custom worker container",
+    )
+    parser.add_argument(
+        "--device",
+        required=False,
+        default="cpu",
+        help="device to run the model on",
+    )
+
+    args, beam_args = parser.parse_known_args()
+
+    config = get_config_for_2b()
+
+    logging.getLogger().setLevel(logging.INFO)
+    beam_options = PipelineOptions(
+        beam_args,
+        streaming=True,
+    )
+    beam_options.view_as(SetupOptions).save_main_session = True
+
+    handler = GemmaPytorchModelHandler(model_variant=args.model_variant,
+                                       checkpoint_path=args.checkpoint_path,
+                                       tokenizer_path=args.tokenizer_path,
+                                       device=args.device)
+
+    with beam.Pipeline(options=beam_options) as p:
+        _ = (
+            p | "Read Topic" >>
+            beam.io.ReadFromPubSub(subscription=args.messages_subscription)
+            | "Parse" >> beam.Map(lambda x: x.decode("utf-8"))
+            | "RunInference-Gemma" >> RunInference(
+                handler)  # Send the prompts to the model and get responses.
+            |
+            "Format Output" >> beam.ParDo(FormatOutput())  # Format the output.
+            | "Publish Result" >> beam.io.gcp.pubsub.WriteStringsToPubSub(
```

Comment:
According to my logs, this method is deprecated since 2.7.0 and should be `WriteToPubSub`

Reply:
Updated